### DISCUSSION TITLE
Is there a Dropwizard 2.1.x EOL date?

### DISCUSSION THREAD

**Discussion by chris-orchard** [dropwizard/dropwizard#7878] [General]
I've been able to find this:

https://github.com/dropwizard/dropwizard/discussions/6930#discussioncomment-5721108
_We were thinking about sunsetting Dropwizard 2.1.x 6 months after the release of Dropwizard 3.0.0 & 4.0.0, so this would be **September 30, 2023**._

But see the latest 2.1.10 release was November 6, 2023, so is still being updated.

How soon is the 2.1.x > 3.x move required?

Also this saying 2.**0**.x EOL was Jan 2023:
https://github.com/dropwizard/dropwizard/discussions/6324

**Comment by joschi** [comment]
> But see the latest 2.1.10 release was November 6, 2023, so is still being updated.

Yes, we totally missed the September 30, 2023 date and December 31, 2023 would be a bit short notice. 😅 

This being said, I'd encourage everybody to invest the small effort needed to migrate to Dropwizard 3.x if possible. Support Dropwizard 2.1.x will eventually end and why not be ahead of the curve?

There is currently no fixed EOL date for Dropwizard 2.1.x but it also doesn't get any support except for dependency updates.

> Also this saying 2.0.x EOL was Jan 2023:

Yes, and this actually happened. 😬

**Reply by chris-orchard** [reply]
Thanks for the info. We're in the process of moving 2.1.x > 3.x anyway but I wanted to know how soon this was required. Seems we'll be ahead of of any EOL.

**Reply by zUniQueX** [reply]
@joschi Should we then announce an EOL for 2.1.x for the end of Jan 2024? I expect the next Jersey 3.1 version being released in a week or two and at that date I'd like to create the `release/5.0.x` branch.

As we start development of Dropwizard 5, I'd like dropping support for 2.1.x.

**Reply by joschi** [reply]
I'm sold. 👍 

Let's set the expiry date for Dropwizard 2.1.x to January 31, 2024.
Given the straightforward migration path, this should be enough time for people to migrate.

[Dropwizard 2.1.0](https://github.com/dropwizard/dropwizard/releases/tag/v2.1.0) was released on May 23, 2022 which means that we supported it for 20 months. That's longer than Spring Boot 2.7.x was supported – [Spring Boot 2.7.0](https://github.com/spring-projects/spring-boot/releases/tag/v2.7.0) was release on May 19, 2022. 😉

**Reply by joschi** [reply]
Here we are: https://github.com/dropwizard/dropwizard/discussions/7880
